src/traces/isosurface/calc.js

```diff
@@ -10,9 +10,19 @@
 
 var colorscaleCalc = require('../../components/colorscale/calc');
 var processGrid = require('../streamtube/calc').processGrid;
+var filter = require('../streamtube/calc').filter;
 
 module.exports = function calc(gd, trace) {
-    trace._len = Math.min(trace.x.length, trace.y.length, trace.z.length, trace.value.length);
+    trace._len = Math.min(
+        trace.x.length,
+        trace.y.length,
+        trace.z.length,
+        trace.value.length
+    );
+
+    ['value', 'x', 'y', 'z'].forEach(function(e) {
+        trace['_' + e] = filter(trace[e], trace._len);
+    });
```

Comment:
Ah, this is ugly. What if I want to know where `_x`, `_y`, `_z`, `_value` are set? With this here, grepping for e.g. `_x` won't help me. 

Please rewrite this as in four almost-identical lines:

```js
trace._value = filter(trace.value, trace._len);
// ...
```

Reply:
Done in https://github.com/plotly/plotly.js/pull/4431/commits/c82da07617b8cac581eea4a4cc291a710683d320.

Reply:
awesome!